### DISCUSSION TITLE
batch delete

### DISCUSSION THREAD

**Discussion by pryme0** [supabase/supabase#4831] [Questions] [answered]
Is there a way to do batch delete with supbase?
Like i pass in an array containing id's of records i want to delete  and it runs the operation for me

**Comment by GaryAustin1** [comment, marked as answer]
With supabase.js or the API delete takes a filter.  for .js you should be able to use .in('id', ['id1', 'id2', 'id3']) where 'id' is your record id.

**Reply by GaryAustin1** [reply]
I edited the above from .contains, to .in.

**Reply by pryme0** [reply]
@GaryAustin1 Your solution worked for me . i am using the Api. Thanks

**Comment by levistringer** [comment]
Searching for this for python, if anyone has found a solution!

**Comment by calebcook7** [comment]
I found another method to be able to bulk delete but it uses a few steps.

1. Create a column in your table named 'delete' and make it a true/false boolean
2.  Javascript to Transform the list of ids/ UUIDs to an array function to get an array function to match for the Upsert matching rows API call.
("id":"UUID", "delete":"true"}
3. Call the upsert matching rows API using the newly formatted array as the array in the JSON data.
(For example [("id":"UUID1", "delete":"true"},{"id":"UUID2", "delete":"true"}, Etc.]
4) Make a function/API call to the delete rows and use the delete column = true as your filter.


This is the only way that I've found to do it.

**Reply by steve-chavez** [reply]
@calebcook7 Hm, why not just use the `delete()` method?

https://supabase.com/docs/reference/javascript/delete

You can pass a range or an array of ids there.
